app/scripts/modules/amazon/src/instance/details/utils.ts

```diff
@@ -17,7 +17,7 @@ export const applyHealthCheckInfoToTargetGroups = (
     if (metric.type === 'TargetGroup') {
       metric.targetGroups.forEach((tg: ITargetGroup) => {
         const group = targetGroups[tg.name] ?? ({} as ITargetGroup);
-        const port = group.healthCheckPort && isNumber(group.healthCheckPort) ? group.healthCheckPort : group.port;
+        const port = parseInt(group.healthCheckPort as any, 10) ?? group.port;
```

Comment:
Can you be more explicit about the type of `group.healthCheckPort` instead of using `any`?

Reply:
The type of `healthCheckPort` in its current form doesn't allow us to use `parseInt`. The core issue here is mismatch of the type definitions between front-end and back-end. In typescript, we are able to provide a more refined definition of `number | 'traffic-port'` whereas, *I guess*,  java doesn't allow such a type definition?

It makes a lot of sense to keep the type as `number | 'traffic-port'` in the front-end, but that makes it hard to use `parseInt`. I could do the following to make it work but I chose the other way since it is more compact.

```js
let port = group.port;

if (group.healthCheckPort != null && group.healthCheckPort !== "traffic-port") {
  port =
    typeof group.healthCheckPort === "string"
      ? parseInt(group.healthCheckPort, 10)
      : group.healthCheckPort;
}
```

I'll update it to the above to make it more explicit to avoid any confusion.